source/module_esolver/esolver_ks.cpp

```diff
@@ -623,6 +630,31 @@ void ESolver_KS<T, Device>::runner(const int istep, UnitCell& ucell)
                                  duration);
 #endif //__RAPIDJSON
 
+        // 12.5) rdmft, add by jghan 2024-04-08/2024-10-09
+        if ( PARAM.inp.ab_initio_type == "rdmft" )
+        {
+            // if ( (!GlobalC::exx_info.info_global.cal_exx && iter == 1) || one_step_exx )
+            if ( !GlobalC::exx_info.info_global.cal_exx || (GlobalC::exx_info.info_global.cal_exx && one_step_exx) )
+            {
+                ModuleBase::matrix occ_number_ks(this->pelec->wg);
+                for(int ik=0; ik < occ_number_ks.nr; ++ik)
+                {
+                    for(int inb=0; inb < occ_number_ks.nc; ++inb) occ_number_ks(ik, inb) /= this->kv.wk[ik];
+                }
+
+                this->update_elec_rdmft(occ_number_ks, *(this->psi));
+
+                //initialize the gradients of Etotal on wg and wfc, and set all elements to 0. 
+                ModuleBase::matrix E_gradient_occNum(this->pelec->wg.nr, this->pelec->wg.nc, true);
+                psi::Psi<T> E_gradient_wfc(this->psi->get_nk(), this->psi->get_nbands(), this->psi->get_nbasis()); 
```

Comment:
would the name of variable be more precise if it were `dedpsi` instead? Plus, although might not be critical, but is it necessary to allocate the memory for all kpoints in-one-shot?

Reply:
ok，I will use dE_dwfc to replace it.

Because my program obtains all k points at once when calculating the gradient, I also do this here.

In addition, E_gradient_occNum and E_gradient_wfc appear here only to verify the correctness of the program. After I write the optimized program, I will delete these two variables from here (the code in esolver_ks.cpp is only for getting the initial value of rdmft calculation after one or several steps of ksdft)